src/vs/workbench/contrib/debug/browser/debugHover.ts

```diff
@@ -178,30 +178,35 @@ export class DebugHoverWidget implements IContentWidget {
 		return this.domNode;
 	}
 
-	async showAt(position: Position, focus: boolean): Promise<void> {
+	async showAt(position: Position, focus: boolean): Promise<void | DEBUG_HOVER_EXPRESSION> {
 		this.showCancellationSource?.cancel();
 		const cancellationSource = this.showCancellationSource = new CancellationTokenSource();
 		const session = this.debugService.getViewModel().focusedSession;
 
 		if (!session || !this.editor.hasModel()) {
 			this.hide();
-			return;
+			return DEBUG_HOVER_EXPRESSION.NOT_AVAILABLE;
 		}
 
 		const result = await this.debugHoverComputer.compute(position, cancellationSource.token);
 		if (this.isVisible() && !result.rangeChanged) {
-			return;
+			return DEBUG_HOVER_EXPRESSION.NOT_CHANGED;
 		}
 
-		if (!result.range || cancellationSource.token.isCancellationRequested) {
+		if (!result.range) {
 			this.hide();
-			return;
+			return DEBUG_HOVER_EXPRESSION.NOT_AVAILABLE;
+		}
+
+		if (cancellationSource.token.isCancellationRequested) {
+			this.hide();
+			return DEBUG_HOVER_EXPRESSION.CANCELLED;
 		}
 
 		const expression = await this.debugHoverComputer.evaluate(session);
-		if (cancellationSource.token.isCancellationRequested || !expression || (expression instanceof Expression && !expression.available)) {
+		if (!expression || (expression instanceof Expression && !expression.available)) {
```

Comment:
You still need to check for cancellation after the await above, it could have been cancelled while that expression was being awaited